DDD.Functions/EloVotingSubmitPair.cs

```diff
@@ -80,6 +80,23 @@ EloVotingConfig eloVoting
                     return new StatusCodeResult((int)HttpStatusCode.BadRequest);
                 }
             }
+            
+            // only bother doing this check if the configuration value is set to greater than 0, otherwise it's disabled
+            if (eloVoting.EloMinimumThinkingTimeSeconds > 0)
+            {
+                var currentTime = keyDates.Now.ToUnixTimeSeconds();
+                var thinkingTime = currentTime - winnerInUnixTimeSeconds;
+
+                // this introduces a "minimum" time that the user has to wait between submitting a vote, we are effectively forcing
+                // the user to slow down and read the entries - as the payload includes the unix time that the pair was generated we 
+                // can get away with using either the winner or loser timestamp.
+                if ((thinkingTime) < eloVoting.EloMinimumThinkingTimeSeconds)
+                {
+                    log.LogWarning("Attempt to submit EloVotingSubmitPar endpoint after {secondsSinceVote} seconds and the minimum allowed is {minimumSeconds}", thinkingTime, eloVoting.EloMinimumThinkingTimeSeconds);
+                    return new StatusCodeResult((int) HttpStatusCode.TooManyRequests);
```

Comment:
If you want users to get any feedback, there will need to be a change to the front end. It's fire-and-forget because any failed submits have either been invalid (eg outside of voting window, mismatched session id) or a resubmit so they're safe to ignore.

For this one the user could just wait a bit longer and it would work, so we might want to stay on the same pair if it fails.

Any preference on that?

Reply:
Nice catch Ben. I wouldn't care for now tbf